kube-derive/src/custom_resource.rs

```diff
@@ -629,6 +633,71 @@ fn generate_hasspec(spec_ident: &Ident, root_ident: &Ident, kube_core: &Path) ->
     }
 }
 
+#[derive(FromField)]
+#[darling(attributes(validated))]
+struct Rule {
+    ident: Option<Ident>,
+    ty: Type,
+    method: Option<Path>,
+    #[darling(multiple, rename = "rule")]
+    rules: Vec<Expr>,
+}
+
+#[derive(FromDeriveInput)]
+#[darling(supports(struct_named))]
+struct CELValidation {
+    #[darling(default)]
+    crates: Crates,
+    data: ast::Data<util::Ignored, Rule>,
```

Comment:
Note: may also be useful to provide naming template for common suffix/prefix per each attribute, in case when `method` is not specified. By default generated method inherits the name of the field for simplicity.